src/cdomains/valueDomain.ml

```diff
@@ -992,6 +992,17 @@ struct
     | `Bot -> BatPrintf.fprintf f "<value>\n<data>\nbottom\n</data>\n</value>\n"
     | `Top -> BatPrintf.fprintf f "<value>\n<data>\ntop\n</data>\n</value>\n"
 
+  let to_yojson = function
+    | `Int n -> `Variant ("Int", Some (ID.to_yojson n))
+    | `Address n -> `Variant ("Address", Some (AD.to_yojson n))
+    | `Struct n -> `Variant ("Struct", Some (Structs.to_yojson n))
+    | `Union n -> `Variant ("Union", Some (Unions.to_yojson n))
+    | `Array n -> `Variant ("Array", Some (CArrays.to_yojson n))
+    | `Blob n -> `Variant ("Blob", Some (Blobs.to_yojson n))
+    | `List n -> `Variant ("List", Some (Lists.to_yojson n))
+    | `Bot -> `Variant ("Bot", None)
+    | `Top -> `Variant ("Top", None)
+
```

Comment:
Based on this, gobview will show all abstract values in base's CPA tagged with their variant name now? I'm just wondering because the currently g2html, which uses the above `printXml` just directly serializes the internal representation and doesn't add a tag to the viewable output. Does this not make the output too verbose?

Reply:
I use the variant names for additional formatting and skip over them otherwise. So, they only make the JSON output more verbose. However, I think the only variant names that I treat differently are `Bot`, `Top`, and `Set`, the last of which was used in the set domain and doesn't seem to exist anymore. So, I guess I might just remove most of these `Variant`s.